src/libclipper/include/clipper/task_executor.hpp

```diff
@@ -469,7 +469,24 @@ class TaskExecutor {
     std::chrono::time_point<std::chrono::system_clock> current_time =
         std::chrono::system_clock::now();
     if (batch_size > 0) {
-      const VersionedModelId &cur_model = keys[0].model_;
+      InflightMessage &first_message = keys[0];
+      const VersionedModelId &cur_model = first_message.model_;
+      const int cur_replica_id = first_message.replica_id_;
+      auto task_latency = current_time - first_message.send_time_;
+      long task_latency_micros =
+          std::chrono::duration_cast<std::chrono::microseconds>(task_latency)
+              .count();
+
+      // Because an RPCResponse is guaranteed to contain data received from
+      // a single model container, the processing container for the first
+      // InflightMessage in the batch is the same processing container
+      // for all InflightMessage objects in the batch
+      std::shared_ptr<ModelContainer> processing_container =
+          active_containers_->get_model_replica(cur_model, cur_replica_id);
+
+      processing_container->update_throughput(batch_size, task_latency_micros);
+      processing_container->latency_hist_.insert(task_latency_micros);
```

Comment:
As it turns out, this is a duplicate call. We already insert the message latency into the containers latency histogram in [`ModelContainer::update_throughput()`](https://github.com/ucbrise/clipper/blob/956f48746d9598fd27c2524a2c01642ea7a024e5/src/libclipper/src/containers.cpp#L46).

Delete this line.

Reply:
Done